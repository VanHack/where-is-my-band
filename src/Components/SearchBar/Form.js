import styled from "styled-components";

const Form = styled.form`
  position: relative;
  display: flex;
  width: 100%;
  justify-content: center;
  align-items: center;
  padding: 15px;

  &:after {
    content: "";
    box-shadow: 0px 0px 25px #000;
    display: block;
    width: 300px;
    max-width: 90%;
    height: 5px;
    position: absolute;
    z-index: -1;
    bottom: 16px;
  }
`;

export default Form;
