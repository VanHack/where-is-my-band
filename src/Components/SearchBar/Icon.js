import styled from "styled-components";

const Icon = styled.button.attrs({
  type: "submit"
})`
  border: none;
  background: #7b1fa2;
  color: white;
  height: 50px;
  line-height: 50px;
  font-size: 20px;
  padding: 0px 15px;
  border-radius: 0px 3px 3px 0px;
`;

export default Icon;
