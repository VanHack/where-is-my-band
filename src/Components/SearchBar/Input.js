import styled from "styled-components";

const Input = styled.input`
  background: white;
  border: none;
  font-size: 16px;
  line-height: 50px;
  height: 50px;
  width: 100%;
  max-width: 300px;
  border-radius: 3px 0px 0px 3px;
  outline: none;
  padding-left: 15px;
`;

export default Input;
