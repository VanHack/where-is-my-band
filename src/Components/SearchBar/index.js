import React, { Component } from "react";
import styled from "styled-components";

const Form = styled.form`
  position: relative;
  display: flex;
  width: 100%;
  justify-content: center;
  align-items: center;

  &:after {
    content: "";
    box-shadow: 0px 0px 25px #000;
    display: block;
    width: 300px;
    max-width: 90%;
    height: 5px;
    position: absolute;
    z-index: -1;
    bottom: 0;
  }
`;

const Input = styled.input`
  background: white;
  border: none;
  font-size: 16px;
  line-height: 50px;
  height: 50px;
  width: 100%;
  max-width: 300px;
  border-radius: 3px 0px 0px 3px;
  outline: none;
  padding-left: 15px;
`;

const Icon = styled.button.attrs({
  type: "submit"
})`
  border: none;
  background: #7b1fa2;
  color: white;
  height: 50px;
  line-height: 50px;
  font-size: 20px;
  padding: 0px 15px;
  border-radius: 0px 3px 3px 0px;
`;

class SearchBar extends Component {
  constructor(props) {
    super(props);
    this.state = { search: "" };

    // Binding functions
    this.handleForm = this.handleForm.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  // controlling the input field
  handleChange(e) {
    this.setState({ search: e.target.value });
  }

  // handling the form submission this will call a
  // function passed as a prop from it's parent component named "search"
  handleForm(e) {
    e.preventDefault();
    this.props.search(this.state.search);
  }

  render() {
    return (
      <Form onSubmit={this.handleForm}>
        <Input
          type="text"
          placeholder="Search"
          value={this.state.search}
          onChange={this.handleChange}
        />
        <Icon className="fa fa-search" aria-hidden="true" />
      </Form>
    );
  }
}

export default SearchBar;
