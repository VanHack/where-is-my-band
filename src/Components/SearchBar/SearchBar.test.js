import React from 'react';
import SearchBar from './index';
import renderer from 'react-test-renderer';

test('renders the component according to snapshot', () => {

  const component = renderer.create(
    <SearchBar search={'Rock Star!'} />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
