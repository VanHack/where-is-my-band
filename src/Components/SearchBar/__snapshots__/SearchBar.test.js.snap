// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component according to snapshot 1`] = `
<form
  className="sc-bdVaJa kOzYEp"
  onSubmit={[Function]}
>
  <input
    className="sc-bwzfXH cIYUei"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
  <button
    aria-hidden="true"
    className="fa fa-search sc-htpNat gTfKmi"
    type="submit"
  />
</form>
`;
